lib/id3c/db/__init__.py

```diff
@@ -60,49 +62,53 @@ def mint_identifiers(session: DatabaseSession, name: str, n: int) -> Any:
     # stochastically.
     max_consecutive_failures = 15
 
-    with session:
-        # Lookup identifier set by name
-        identifier_set = session.fetch_row("""
-            select identifier_set_id as id
-              from warehouse.identifier_set
-             where name = %s
-            """, (name,))
+    # Lookup identifier set by name
+    identifier_set = session.fetch_row("""
+        select identifier_set_id as id
+          from warehouse.identifier_set
+         where name = %s
+        """, (name,))
 
-        if not identifier_set:
-            LOG.error(f"Identifier set «{name}» does not exist")
-            raise IdentifierSetNotFoundError(name)
+    if not identifier_set:
+        LOG.error(f"Identifier set «{name}» does not exist")
+        raise IdentifierSetNotFoundError(name)
 
-        while len(minted) < n:
-            m = len(minted) + 1
+    started = datetime.now()
 
-            LOG.debug(f"Minting identifier {m}/{n}")
+    while len(minted) < n:
+        m = len(minted) + 1
 
-            try:
-                with session.savepoint(f"identifier {m}"):
-                    new_identifier = session.fetch_row("""
-                        insert into warehouse.identifier (identifier_set_id, generated)
-                            values (%s, now())
-                            returning uuid, barcode, generated
-                        """,
-                        (identifier_set.id,))
+        LOG.debug(f"Minting identifier {m}/{n}")
 
-                    minted.append(new_identifier)
+        try:
+            with session.savepoint(f"identifier {m}"):
+                new_identifier = session.fetch_row("""
+                    insert into warehouse.identifier (identifier_set_id, generated)
+                        values (%s, now())
+                        returning uuid, barcode, generated
+                    """,
+                    (identifier_set.id,))
 
-            except ExclusionViolation:
-                LOG.debug("Barcode excluded")
+                minted.append(new_identifier)
 
-                failures.setdefault(m, 0)
-                failures[m] += 1
+        except ExclusionViolation:
+            LOG.debug("Barcode excluded")
 
-                if failures[m] > max_consecutive_failures:
-                    LOG.error("Too many consecutive failures, aborting")
-                    raise TooManyFailuresError(failures[m])
-                else:
-                    LOG.debug("Retrying")
+            failures.setdefault(m, 0)
+            failures[m] += 1
+
+            if failures[m] > max_consecutive_failures:
+                LOG.error("Too many consecutive failures, aborting")
+                raise TooManyFailuresError(failures[m])
+            else:
+                LOG.debug("Retrying")
+
+    duration = datetime.now() - started
+    rate = duration.total_seconds() / n
```

Comment:
When I see 'rate' I think barcodes per second, so I think n / duration.total_seconds().
A variable name like secs_per_barcode or something like that would match better what you're capturing.

Reply:
Yeah, that's fair. I had it as a proper rate initially but then switched to time per barcode and forgot to update the variable name. I'll fix this up.

Reply:
Done in the repush, and both rates are now logged. Look good?

Reply:
That's cool that you're logging both. Both will be handy.